### DISCUSSION TITLE
useUserData() no loading state?

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#2729] [Q&A] [answered]
I'm curious why useUserData() doesn't have any loading state? Just using ! in front of the constant here, doesn't work well in some situatinos on the frontend (React). I couldn't find any other hooks to get user information that had loading. What is the best way to handle this client side?

**Comment by dbarrosop** [comment, marked as answer]
user information is read from local state so there is no async call, that's why there is no `isLoading`. If you have no data under some circumstances I am guessing it might be because the user hasn't been authenticated yet and there is no data in the local state yet so you can use isLoading from the authentication hook to wait to read user information too.

**Reply by osseonews** [reply]
Thanks. Yeah we are checking first with  useAuthenticationStatus(), so you are right, if they are authenticated the user information is just there. Didn't realize it was not async
